service/storage_proxy.cc

```diff
@@ -1726,11 +1726,17 @@ class storage_proxy::cancellable_write_handlers_list : public bi::list<abstract_
         // handler is being removed from the b::list, so if any live iterator points at it,
         // move it to the next object (this requires that the list is traversed in the forward
         // direction).
+        bool drop_end = false;
         for (auto& itp : _live_iterators) {
             if (&**itp == handler) {
                 ++*itp;
+                drop_end |= (*itp == end());
             }
         }
+        if (drop_end) {
```

Comment:
Since there's only one item in the list, why bother with drop_end?

Reply:
There can be more theoretically if multiple cancellations is running. Why is it a vector in the first place?

Reply:
Overkill

Reply:
What do you mean?

Reply:
probably we don't need to memorize if we have `end()` iterator in the vector? in other words, we can just go ahead and do the `remove_if()` and `erase()` dance?

Reply:
Well, I need to know what the "Overkill" means before I do yet another version based on a guesse. May be this is a reply to "Why is it a vector in the first place?". In this case it is not an overkill. We may have several cancel_write_handlers running in parallel.

Reply:
> What do you mean?

The author of the patch that introduced the vector planned it to be usable by more users, but it never happened.

Reply:
I think the vector is used. See https://github.com/scylladb/scylladb/pull/20977#discussion_r1791633308